File: send_udp.cpp
#include <iostream>
#include <stdio.h>
#include <sys/socket.h>
#include <netinet/in.h>
#include <arpa/inet.h>
#include <unistd.h>
#include <strings.h>
#include <string.h>
#include <fcntl.h>
#include <stdlib.h>
#include <pthread.h>
#include <string>

using namespace std;

int PORT = 80;  
int toggle = 0;

bool udpSend(const char *msg){
    
    int i, j, k, l;
    for(i=10;i<=200;i++) {
        for(j = 10;j<=200;j++) {
            for(k=10;k<=200;k++) {
                for(l=10;l<=200;l++) {
                    sockaddr_in servaddr;
                    int fd = socket(AF_INET,SOCK_DGRAM,0);
                    if(fd<0){
                        perror("cannot open socket");
                        return false;
                    }
                    //string s = ""; s += to_string(i) +"."+ to_string(j) +"."+ to_string(k) +"."+ to_string(l);
                 string s = "10.117.147.138";    
		
		const char *HOSTNAME = s.c_str();
                    //cout<<"IP: "<<s<<" "<<HOSTNAME<<endl;

                    bzero(&servaddr,sizeof(servaddr));
                    servaddr.sin_family = AF_INET;
                    servaddr.sin_addr.s_addr = inet_addr(HOSTNAME);
                    servaddr.sin_port = htons(PORT);
                   
                    if (sendto(fd, msg, strlen(msg)+1, 0, // +1 to include terminator
                               (sockaddr*)&servaddr, sizeof(servaddr)) < 0){
                        perror("cannot send message");
                        close(fd);
                        return false;
                    }
                    close(fd);
                }
            }
        }
    }
    
    return true;
}


void *perpetual(void *vargp) {
    while(1) {
        udpSend("Hello World");
    }
}

int main() {
    pthread_t tid1, tid2, tid3, tid4;
    int i = 0;
    pthread_create(&tid1, NULL, perpetual, NULL);
    pthread_create(&tid2, NULL, perpetual, NULL);
    pthread_create(&tid3, NULL, perpetual, NULL);
    pthread_create(&tid4, NULL, perpetual, NULL);
    pthread_join(tid1, NULL);
    pthread_join(tid2, NULL);
    pthread_join(tid3, NULL);
    pthread_join(tid4, NULL);
    
}

File: send_udp.py
import socket
import os
import sys
import urllib
import random
import threading


def get_random_ip():
	a = random.randint(10, 220)
	b = random.randint(10, 220)
	c = random.randint(10, 220)
	d = random.randint(10, 220)
	randomip = ".".join([str(x) for x in [a, b, c, d]])
	return randomip

	#randomip = "http://"+randomip
	#return randomip

def send_udp(threadnum):
	i = 0
	while(1):
		i += 1
		UDP_IP = get_random_ip()
		UDP_PORT = 8080
		MESSAGE = "Hello, World!"

		if(i%100000==0): print("IP: {0} Thread: {1}".format(UDP_IP, threadnum))

		#print "UDP target IP:", UDP_IP
		#print "UDP target port:", UDP_PORT
		#print "message:", MESSAGE

		sock = socket.socket(socket.AF_INET, # Internet
		             socket.SOCK_DGRAM) # UDP
		sock.sendto(MESSAGE, (UDP_IP, UDP_PORT))
		sock.close()


numthreads = 4

for i in range(0, numthreads):
	threading.Thread(target=send_udp, args=(i,)).start()

#send_udp()